app/main.py

```diff
@@ -1,23 +1,31 @@
 import init_django_orm  # noqa: F401
 
-from django.db.models import QuerySet
+from django.db.models import QuerySet, Q, F
+from db.models import Customer, LoyaltyProgram, LoyaltyProgramParticipant
 
 
 def all_loyalty_program_names() -> QuerySet:
-    pass
+    return LoyaltyProgram.objects.values_list("name", "bonus_percentage")
 
 
 def not_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.filter(
+        last_activity__range=["2021-01-02", "2021-12-31"]
+    ).values("customer__first_name")
 
 
 def most_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.values_list(
+        "customer__first_name", "customer__last_name", "sum_of_spent_money"
+    ).order_by("-sum_of_spent_money")[:5]
 
 
 def clients_with_i_and_o() -> QuerySet:
-    pass
+    return Customer.objects.filter(
+        Q(first_name__startswith="I") | Q(last_name__contains="o"))
 
 
 def bonuses_less_then_spent_money() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.values(
```

Comment:
Might be my personal preference, but it seems more readable to place `.filter()` before `.values()`.